tests/integration/640.command.recipes.test.cjs

```diff
@@ -96,3 +97,46 @@ test('Does not generate a new VS Code settings file if the user does not confirm
     t.is(error.code, 'ENOENT')
   })
 })
+
+test.only('Handles JSON with comments', async (t) => {
+  await withSiteBuilder('repo', async (builder) => {
+    const comment = '// sets value for someSetting'
+    await builder
+      .withContentFile({
+        path: '.vscode/settings.json',
+        content: `{
+          "deno.enablePaths":["/some/path"],
+          "someSetting":"value" ${comment}
+        }`,
+      })
+      .buildAsync()
+
+    const childProcess = execa(cliPath, ['recipes', 'vscode'], {
+      cwd: builder.directory,
+    })
+    const settingsPath = resolve(builder.directory, '.vscode', 'settings.json')
+
+    handleQuestions(childProcess, [
+      {
+        question: `There is a VS Code settings file at ${settingsPath}. Can we update it?`,
+        answer: answerWithValue(CONFIRM),
+      },
+      {
+        question: 'The Deno VS Code extension is recommended. Would you like to install it now?',
+        answer: answerWithValue(NO),
+      },
+    ])
+
+    await childProcess
+
+    const settingsText = await readFile(`${builder.directory}/.vscode/settings.json`, { encoding: 'utf8' })
+    t.log(settingsText)
+    t.true(settingsText.includes(comment))
+
+    const settings = parse(settingsText, null, true)
+    t.is(settings.someSetting, 'value')
+    t.is(settings['deno.enable'], true)
+    t.is(settings['deno.importMap'], '.netlify/edge-functions-import-map.json')
+    t.deepEqual([...settings['deno.enablePaths']], ['/some/path', 'netlify/edge-functions'])
```

Comment:
Spreading the array here since that converts it from the [CommentArray](https://www.npmjs.com/package/comment-json#commentarray) type to a native JS array for comparison.